src/redirect.rs

```diff
@@ -179,6 +182,16 @@ pub fn check_redirect(policy: &RedirectPolicy, next: &Url, previous: &[Url]) ->
     }).inner
 }
 
+pub fn remove_sensitive_headers(mut headers: Headers, next: &Url, previous: &[Url]) -> Headers {
+    let cross_host = next.host().unwrap() != previous.last().unwrap().host().unwrap();
```

Comment:
Options can be compared also, so I'd probably leave the unwrap out.

Also, this is far better than the nothing that was there before. I think we can also check for subdomains, but if you'd rather, we can just file a new issue regarding that.